File: sentinel/AddTwoNumbers.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        // construct the sum terms
        // get the sum
        // convert the sum to a linked list
        
        // initialize result, terms and sum
        ListNode result = new ListNode(0);
        ListNode sentinel = result;
        int rest = 0;
        
        while (l1 != null || l2 !=null){
            
            int a = (l1 != null) ? l1.val : 0;
            int b = (l2 != null) ? l2.val : 0;   
            
            int sum = a + b + rest;   
            
            // if sum is > 10
            if(sum >= 10){
                rest = sum/10;
                sum = 0;
            }
            
            result.val = sum;            
            result.next = new ListNode(0);
            result = result.next;
            
            l1 = l1.next;
            l2 = l2.next;
            
        }
        return sentinel;     
    }
}

File: sentinel/AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        result = ListNode(0)
        sentinel = result
        rest = 0
        while l1 is not None or l2 is not None or rest>0:
            a = l1.val if l1 is not None else 0
            b = l2.val if l2 is not None else 0
            sum = a + b + rest
            rest = int(sum/10)
            result.next = ListNode(sum%10)
            result = result.next
            if l1 is not None: 
                l1 = l1.next
            if l2 is not None:
                l2 = l2.next
        return sentinel.next       
        